redisai/commands.go

```diff
@@ -327,17 +327,58 @@ func (c *Client) DagRunRO(loadKeys []string, dagCommandInterface DagCommandInter
 	return dagCommandInterface.ParseReply(reply, err)
 }
 
+// DagExecute Direct acyclic graph of operations to run within RedisAI
+func (c *Client) DagExecute(loadKeys, persistKeys []string, routing string, timeout int64, dagCommandInterface DagCommandInterface) ([]interface{}, error) {
+	commandArgs, err := dagCommandInterface.FlatArgs()
+	if err != nil {
+		return nil, err
+	}
+	args := AddDagExecuteArgs(loadKeys, persistKeys, routing, timeout, commandArgs)
+	reply, err := c.DoOrSend("AI.DAGEXECUTE", args, nil)
+	return dagCommandInterface.ParseReply(reply, err)
+}
+
+// DagExecuteRO is the read-only variant of DagExecute
+func (c *Client) DagExecuteRO(loadKeys []string, routing string, timeout int64, dagCommandInterface DagCommandInterface) ([]interface{}, error) {
+	commandArgs, err := dagCommandInterface.FlatArgs()
+	if err != nil {
+		return nil, err
```

Comment:
I can't figure out how to cover this error. `FlatArgs` always returns nil as an error...

Reply:
But why the `FlatArgs` method always returns `nil`? Consider removing the second return value from this method.